doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,318 @@
+
+### 1.2.4 Event logging
+**This feature will provide event logging for:** loopback action set on IP interface.
+
+###### Table 1: Event logging
+| Event                                     | Severity |
+|:------------------------------------------|:---------|
+| IP interface loopback action set: success | NOTICE   |
+| IP interface loopback action set: error   | ERROR    |
+
+# 2 Design
+## 2.1 Overview
+Intfmgrd subscribes for changes in L3 interface tables VLAN_INTERFACE, INTERFACE and PORTCHANNEL_INTERFACE in config DB. Whenever there is a change in one of these tables, intfmgrd is notified. Intfmgrd then updates INTF_TABLE table in app DB. Upon change in INTF_TABLE, intforch is notified.
+Intforch performs validity checks, call the relevant SAI API and updates internal cache. There is an existing mechanizm in intforch that postpones handling of changes in INTF_TABLE until interface is ready, we will use it.
+
+## 2.2 DB schema
+### 2.2.1 Config DB
+To support loopback action set on L3 interface, a new field-value pair is defined in CONFIG_DB in the following tables.
+Valid values are "drop" or "forward".
+
+1. VLAN_INTERFACE
+```
+VLAN_INTERFACE|interface-name
+    "loopback_action": action (OPTIONAL) 
+```
+Example:
+```
+"VLAN_INTERFACE": {
+    "Vlan100": {
+        "loopback_action": "drop",
+        "mac_addr": "00:01:02:03:04:10",
+        "mpls": "enable"
+    },
+}
+```
+
+2. INTERFACE
+```
+INTERFACE|interface-name
+    "loopback_action": action 
+```
+Example:
+
+```
+"INTERFACE": {
+    "Ethernet248": {
+        "loopback_action": "forward",
+        "mac_addr": "00:01:02:03:04:11",
+        "mpls": "enable"
+    },
+}
+```
+3. PORTCHANNEL_INTERFACE
+```
+PORTCHANNEL_INTERFACE|interface-name
+    "loopback_action": action 
+```
+Example:
+```
+"PORTCHANNEL_INTERFACE": {
+    "PortChannel1": {
+        "loopback_action": "drop",
+        "mac_addr": "00:01:02:03:04:12",
+        "mpls": "enable"
+    },
+}
+```
+
+### 2.2.2 App DB
+To support loopback action set on L3 interface, a new field-value pair is defined in APP DB in table INTF_TABLE.
+```
+INTF_TABLE|interface-name
+    "loopback_action": action
+```
+Example for interface vlan:
+```
+    "INTF_TABLE": {
+        "Vlan100": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:10",
+            "mpls": "enable"            
+        }
+    }
+```
+Example for interface Ethernet:
+```
+    "INTF_TABLE": {
+        "Ethernet248": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:11",
+            "mpls": "enable"
+        }
+    }
+```
+Example for interface port-channel:
+```
+    "INTF_TABLE": {
+        "PortChannel1": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:12",
+            "mpls": "enable"            
+        }
+    }
+```
+
+### 2.2.3 Configuration migration
+No special handling is required
```

Comment:
i suggest to exaplin what will be the behaviour when upgrading from sw image which does not support this new schema to a one which supports and what happenes when the loopback action is not set